src/uu/date/src/date.rs

```diff
@@ -103,6 +107,75 @@ enum Iso8601Format {
     Ns,
 }
 
+// Enum for system clock resolution
+enum TimeResolution {}
+const RESOLUTION_DIVISOR: f32 = 1_000_000_000.0;
+
+impl TimeResolution {
+    fn generate() -> f32 {
+        // Getting time from system
+        let time = std::time::SystemTime::now();
+
+        let duration = time
+            .duration_since(std::time::UNIX_EPOCH)
+            .unwrap_or_default();
+
+        let secs: u64 = duration.as_secs();
+        let nsecs: u32 = duration.subsec_nanos();
+
+        let mut ts: timespec = timespec {
+            tv_sec: secs as c_long,
+            tv_nsec: nsecs as c_long,
+        };
+
+        let resolution: f32 = match Self::clock_type(&mut ts) {
+            Some(CLOCK_REALTIME) => {
+                let mut res_ts = ts;
+                unsafe { libc::clock_getres(CLOCK_REALTIME, &mut res_ts) };
+                let resolution_secs: f32 =
+                    res_ts.tv_sec as f32 + res_ts.tv_nsec as f32 / RESOLUTION_DIVISOR;
+                resolution_secs
+            }
+            Some(CLOCK_MONOTONIC) => {
+                let mut res_ts: timespec = ts;
+                unsafe { libc::clock_getres(CLOCK_MONOTONIC, &mut res_ts) };
```

Comment:
`clock_getres` can fail, and the return value must be checked: https://docs.rs/libc/latest/libc/fn.clock_getres.html

And because it must be checked anyway, I don't understand why this code calls `SystemTime::now();` and then also `clock_gettime`.